### PR TITLE
feat: add action object

### PR DESCRIPTION
**Pull request by love98ooo**
Separate actions from rules as an object.

### CODE REVIEW

File: object/action.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+)
+
+type Action struct {
+	Owner         string `xorm:"varchar(100) notnull pk" json:"owner"`
+	Name          string `xorm:"varchar(100) notnull pk" json:"name"`
+	CreatedTime   string `xorm:"varchar(100) notnull" json:"createdTime"`
+	UpdatedTime   string `xorm:"varchar(100)" json:"updatedTime"`
+	Type          string `xorm:"varchar(100) notnull" json:"type"`
+	StatusCode    int    `xorm:"int notnull" json:"statusCode"`
+	ImmunityTimes int    `xorm:"int notnull" json:"immunityTimes"` // minutes
```
Comment (nomeguy) [inline object/action.go:30]: Let GPT give you a good English name

File: controllers/action.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func (c *ApiController) AddAction() {
+	if c.RequireSignedIn() {
+		return
+	}
+
+	time := util.GetCurrentTime()
+	action := object.Action{
+		CreatedTime: time,
```
Comment (nomeguy) [inline controllers/action.go]: Copy paste from other object, don't write by yourself

File: object/action_cache.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+package object
+
+import (
+	"fmt"
+
+	"github.com/casbin/caswaf/util"
+)
+
+var actionMap = map[string]*Action{}
```
Comment (nomeguy) [inline object/action_cache.go:23]: Why action still needs actionMap? Delete all

**Review by nomeguy — CHANGES_REQUESTED** [review]

File: object/action.go (hunk 1 above)
Comment (love98ooo) [reply object/action.go:30]: This is what AWS uses. I didn't make it up.

image

File: object/action_cache.go (hunk 3 above)
Comment (love98ooo) [reply object/action_cache.go:23]: If an action updates, we need the map to make it work immediately.

File: controllers/action.go (hunk 2 above)
Comment (love98ooo) [reply controllers/action.go]: image

It is copied from controller/rule.go

**Review by nomeguy — APPROVED** [review]

**Pull request merged**